### ISSUE TITLE
envkey does not update secrets in unicorn when reloaded with USR2

### ISSUE THREAD

**Issue by sethpollack** [envkey/envkey-ruby#5]
Basically the first time unicorn boots, envkey loads the env vars but does not keep track of which keys it set. Then on subsequent reloads with `USR2` it fails to update the values because they are now already set and envkey will not override them.

**Comment by danenania** [comment]
Looking into this--we probably need to handle unicorn reloads in the same way as spring reloads.

**Comment by sethpollack** [comment]
Yeah, or expose a hook that users can invoke in `after_fork` in `unicorn.rb`, `resque.rb` etc...

**Comment by danenania** [comment]
Hmm, actually you should already be able to do that by requiring `envkey/core` and calling `Envkey::Core.load_env` yourself. It returns two arrays, the keys set via dotenv and via envkey, so that it knows to override just the envkey ones on subsequent loads. I think this should work in `unicorn.rb`:

```
require "envkey/core"

ts = Time.now
overload_dotenv_vars, overload_envkey_vars = Envkey::Core.load_env
after_fork do
   Envkey::Core.load_env(ts, (overload_dotenv_vars || []), (overload_envkey_vars || []))
end
```

This interface can be cleaned up a bit and documented, but does that work for you in the meantime?

**Comment by sethpollack** [comment]
I'll give it a try thanks!

**Comment by sethpollack** [comment]
Doesn't seem to work, probably because the Railtie already ran?

**Comment by danenania** [comment]
Ah yes, in your `Gemfile` try adding `require: false`:

`gem 'envkey', require: false`

That should prevent the Railtie from running.

**Comment by sethpollack** [comment]
This doesn't work for me, the unicorn file gets loaded every time and resets `overload_dotenv_vars` and `overload_envkey_vars`.

**Comment by sethpollack** [comment]
What about doing something like this in `Envkey::Core`?

```  
def self.envkey_original_env
    ENV["ENVKEY_ORIGINAL_ENV"] ||= ENV.to_json
    JSON.parse(ENV["ENVKEY_ORIGINAL_ENV"]).to_h
end
```

**Comment by danenania** [comment]
Yeah something along those lines should work--will get it pushed up soon!

**Comment by sethpollack** [comment]
Cool, thanks!

**Comment by danenania** [comment]
Should now be fixed in 1.2.6!

**Comment by sethpollack** [comment]
Thanks!
